plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-shipping-address-block/block.tsx

```diff
@@ -53,7 +53,16 @@ const Block = (): JSX.Element => {
 			delete syncValues.company;
 		}
 
-		setBillingAddress( syncValues );
+		// Only sync if any values are different
+		const needsSync = ! Object.keys( syncValues ).every(
+			( key ) =>
+				syncValues[ key as keyof BillingAddress ] ===
```

Comment:
Some things come to mind when checking for billing address object equality:
- do we have arrays or objects for the value, because deep equality won't work; I'm thinking country filed, maybe additional fields?
- you can check for keys length and return true if they are not equal, a small optimization
-

Reply:
addresses are flat objects so a quick surface level iteration on them is fine. additional fields are top level values here.

Reply:
Would there be value in abstracting this into a helper somewhere?

Reply:
It's only used there so no need?

Reply:
thank you Nadir for confirming those they are flat objects. Did you consider optimizing based on the key length check?
```
  if (Object.keys(syncValues).length !== Object.keys(billingAddress).length) {
            return true;
        }
       ```